doc/fr/developper/writestreamconnector.rst

```diff
@@ -0,0 +1,586 @@
+    local service_description = broker_cache:get_service_description(e.host_id, e.service_id)
+  
+    -- what if we could not get them from cache
+    if not host_name then
+      broker_log:warning(1, "event_queue:add: host_name for id " .. e.host_id .. " not found. Restarting centengine should fix this.")
+      host_name = e.host_id
+    end
+    if not service_description then
+      broker_log:warning(1, "event_queue:add: service_description for id " .. e.host_id .. "." .. e.service_id .. " not found. Restarting centengine should fix this.")
+      service_description = e.service_id
+    else
+      service_description = service_description:gsub(" ", "_")
+    end
+  
+    -- we finally append the event to the events table
+    broker_log:info(3, "event_queue:add: adding \"" .. service_description..",host="..host_name.." "..metric:gsub(" ", "_").."="..e.value.." "..e.ctime.."000000000\" to event list.")
+    self.events[#self.events + 1] = "'"..service_description.."',host='"..host_name.."' '"..metric:gsub(" ", "_").."'="..e.value.." "..e.ctime.."000000000\n"
+  
+    -- then we check whether it is time to send the events to the receiver and flush
+    if #self.events >= self.max_buffer_size then
+      broker_log:info(2, "event_queue:add: flushing because buffer size reached " .. self.max_buffer_size .. " elements.")
+      self:flush()
+      return true
+    elseif os.time() - self.__internal_ts_last_flush >= self.max_buffer_age then
+      broker_log:info(2, "event_queue:add: flushing " .. #self.events .. " elements because buffer age reached " .. (os.time() - self.__internal_ts_last_flush) .. "s and max age is " .. self.max_buffer_age .. "s.")
+      self:flush()
+      return true
+    else
+      return false
+    end
+  end
+
+A function to flush the queue
+*****************************
+
+Once the events added in the queue and the maximum size of the queue or the 
+timeout is reached, events will be sent to the InfluxDB storage. This function 
+will build data from the queue and send these to the storage. If an error 
+occurs, it will be write into the associated log file:
+
+.. code-block:: lua
+
+  function event_queue:flush()
+    broker_log:info(2, "event_queue:flush: Concatenating all the events as one string")
+    --  we concatenate all the events
+    local http_post_data = ""
+    local http_result_body = {}
+    for i, raw_event in ipairs(self.events) do
+      http_post_data = http_post_data .. raw_event
+    end
+    broker_log:info(2, "event_queue:flush: HTTP POST request \"" .. self.http_server_protocol .. "://" .. self.http_server_address .. ":" .. self.http_server_port .. "/write?db=" .. self.influx_database .. "\"")
+    broker_log:info(3, "event_queue:flush: HTTP POST data are: '" .. http_post_data .. "'")
+  
+    -- build url
+    local influxdb_url = self.http_server_protocol.."://"..self.http_server_address..":"..self.http_server_port.."/write?db="..self.influx_database
```

Comment:
Your coding style is strange, sometimes there are spaces around ".." and sometimes not.